modules/repository-datastax/src/main/java/com/intuit/wasabi/repository/cassandra/accessor/ExperimentTagAccessor.java

```diff
@@ -31,15 +32,15 @@
 public interface ExperimentTagAccessor {
 
     @Query("select * from experiment_tag where app_name = ?")
-    ListenableFuture<Result<ExperimentTagsByApplication>> getExperimentTagsAsync(String applicationNames);
+    ListenableFuture<Result<ExperimentTagsByApplication>> getExperimentTagsAsync(String appName);
 
-    @Query("insert into experiment_tag(app_name, tags) values (?,?)")
-    void insertByApp(String appName, Set<String> tags);
+    @Query("insert into experiment_tag(app_name, exp_id, tags) values (?,?,?)")
+    void insert(String appName, UUID experimentId, Set<String> tags);
 
-    @Query("update experiment_tag set tags = tags + ? WHERE app_name = ?")
-    Statement update(Set<String> tags, String appName);
+    @Query("update experiment_tag set tags = tags + ? WHERE app_name = ? and exp_id = ?")
+    Statement update(Set<String> tags, String appName, UUID experimentId);
```

Comment:
Also, I have a question regarding these methods. I saw that we are completely rewriting the tag list when we add or remove tags. Can we get rid of these methods then?